jobs/garbage-collection.Jenkinsfile

```diff
@@ -56,13 +56,17 @@ lock(resource: "gc-${params.STREAM}") {
             def originalTimestamp = originalBuildsJson.timestamp
             def acl = pipecfg.s3.acl ?: 'public-read'
 
-            withCredentials([file(variable: 'GCP_KOLA_TESTS_CONFIG', credentialsId: 'gcp-image-upload-config')]) {
+            withCredentials([
+                file(variable: 'GCP_KOLA_TESTS_CONFIG', credentialsId: 'gcp-image-upload-config'),
+                file(variable: 'REGISTRY_SECRET', credentialsId: 'oscontainer-push-registry-secret')
+            ]) {
                 stage('Garbage Collection') {
                     pipeutils.shwrapWithAWSBuildUploadCredentials("""
                     cosa coreos-prune --policy ${new_gc_policy_path} \
                     --stream ${params.STREAM} ${dry_run} \
                     --gcp-json-key=\${GCP_KOLA_TESTS_CONFIG} \
                     --acl=${acl} \
+                    --registry-auth-file=\${REGISTRY_SECRET} \
                     --aws-config-file \${AWS_BUILD_UPLOAD_CONFIG}
```

Comment:
This isn't new in this PR, but I noticed it..

It's a bit odd to pass some secrets one way and some another.. maybe we should consider instead of using `shwrapWithAWSBuildUploadCredentials()` here to just explicitly mention the `AWS_BUILD_UPLOAD_CONFIG` one in the block above too.